airbyte_cdk/sources/types.py

```diff
@@ -139,3 +148,6 @@ def __ne__(self, other: Any) -> bool:
 
     def __json_serializable__(self) -> Any:
         return self._stream_slice
+
+    def __hash__(self) -> int:
+        return hash(orjson.dumps(self._stream_slice))
```

Comment:
Is orjson ensuring order of fields? If so, what is the order? Note that we have `SliceEncoder` in `adapters.py` which seems to indicate that we have slices that would need more than just a `dumps` to work

Reply:
> Is orjson ensuring order of fields? If so, what is the order?

what do you mean by order? dict is already an ordered structure, we can pass `orjson.OPT_SORT_KEYS` to forcefully sort dict by dict_keys.

But I'm ok to use standart json library here, this is not a heavily used operation.

Reply:
Dict are ordered by when they were inserted. It means that dictionaries that are equal might not generate the same JSON representation. For example:
```
import json
x = {}
x[1] = 1
x[2] = 2
y = {}
y[2] = 2
y[1] = 1

print(f"JSON compare: {json.dumps(x) == json.dumps(y)}")
print(f"Python compare: {x == y}")
```

... will output:
```
JSON compare: False
Python compare: True
```

To fix the ordering problem, we would need to sort the keys.

Reply:
gotcha, sort added